book/src/terminology.md

```diff
@@ -18,6 +18,10 @@ The result of interpreting all programs on the ledger at a given [tick height](t
 
 A contiguous set of [entries](terminology.md#entry) on the ledger covered by a [vote](terminology.md#ledger-vote). A [leader](terminology.md#leader) produces at most one block per [slot](terminology.md#slot).
 
+## blockhash
+
+The hash over a bank's state upon freeze.  Used to historically identify the block originated from the bank.
```

Comment:
We don't define "freeze".  Perhaps replace with "upon completion of a block"

Reply:
Just realized this definition isn't even correct.  We use only the hash of the last entry, which I'm not sure is the RightThing(TM) to do.  Seems like we should be covering more of the bank state than that.  Is this a performance thing or can it be changed?

Reply:
Sorry, I wasn't clear.  By _covering_ I meant the blockhash should probably be generated using more of the bank state.  This would, a) make its ID unambiguous, and b) allow verification of the included account states.  I'm not sure of the intention of using only the `EntryId`

Reply:
How about: "A preimage resistant hash of the ledger state at a given block height."  Some other part of the book can explain how it's calculated and/or how it's used. It might say, "A blockhash is calculated from a SHA256 hash of the previous blockhash and all changes to the ledger state since the previous block."

Reply:
Yes, that's a good definition.

It looks like I'm letting responsibilities meld a bit here... What we store in the `blockhash_queue` at https://github.com/solana-labs/solana/blob/master/runtime/src/bank.rs#L800-L804 is not the "blockhash" as is calculated at https://github.com/solana-labs/solana/blob/master/runtime/src/bank.rs#L592-L605.  The latter is what I would consider to be a "blockhash" so it's confusing that it's not what's going into the thing called `blockhash_queue`

Reply:
Mistake on my end. I mixed `Blockhash` and `BankHash`.  The blockhash is not a hash of the ledger state. It's only a hash of the transactions (specifically, the fee payer's signature).  Vote transactions tie together a blockhash and a bank hash.

Reply:
So... "A preimage resistant hash of the ledger at a given block height."

Reply:
Ok, yeah let's go with that.  I'm still unclear as to why we don't use the `BankHash` as the `BlockHash`, but that's out side the scope of this PR